### DISCUSSION TITLE
difference between Restore Settings and Reset to Defaults

### DISCUSSION THREAD

**Discussion by tome9111991** [Jyers/Marlin#600] [Q&A] [answered]
Hello
what is the difference between Restore Settings and Reset to Defaults?

 Reset to Defaults set my EEPROM to firmware setting.

but Restore Settings i see no effect

**Comment by gothcha** [comment]
Hi,  Restore settings actually means re-store settings, maybe the text should be changed to make it clearer?
  
This is the same as Save settings and puts the current data into EEPROM

**Reply by gothcha** [reply]
Apologies my previous comments were wrong, the Restore option ‘Loads’ the settings from EEPROM; Save puts the current memory into EEPROM and as you said Reset writes the values from the config file to EEPROM. I also find it confusing  (as you can see) I modified my code to say Load instead of Restore which to me makes more sense!

**Reply by Cal-LF** [reply]
Where do you modify the text for the menus?

**Reply by gothcha** [reply]
creality_dwin.cpp line 1689 (or there about)

Draw_Menu_Item(row, ICON_ReadEEPROM, (char*)"Restore Settings");

**Comment by Schnabulation** [comment, marked as answer]
Restore Settings: Loads the settings from EEPROM into RAM.

Reset to Defaults: Loads the settings compiled into the firmware (configuration.h and configuration_adv.h) into RAM.

Save settings: Stores the settings from RAM into EEPROM.

Important: When you power-cycle the printer it will always load the settings from EEPROM into RAM. If you change something and don't store it into EEPROM it will be gone after the next power-cycle.
